packages/zip/src/buffer-writer.js

```diff
@@ -24,7 +24,7 @@ const getPrivateFields = self => {
 };
 
 const assertNatNumber = n => {
-  if (Number.isSafeInteger(n) && n >= 0) {
+  if (Number.isSafeInteger(n) && /** @type {number} */ (n) >= 0) {
```

Comment:
/me disappointed that TS not inferring from `Number.isSafeInteger`.

Reply:
Me too.